src/algebra/order/upper_lower.lean

```diff
@@ -34,13 +34,10 @@ section ordered_comm_group
 variables {α : Type*} [ordered_comm_group α] {s t : set α} {a : α}
 
 @[to_additive] lemma is_upper_set.smul (hs : is_upper_set s) : is_upper_set (a • s) :=
-begin
-  rintro _ y hxy ⟨x, hx, rfl⟩,
-  exact mem_smul_set_iff_inv_smul_mem.2 (hs (le_inv_mul_iff_mul_le.2 hxy) hx),
-end
+hs.image $ order_iso.mul_left _
 
 @[to_additive] lemma is_lower_set.smul (hs : is_lower_set s) : is_lower_set (a • s) :=
-hs.of_dual.smul
+hs.image $ order_iso.mul_left _
```

Comment:
What's wrong with the old proof? (or `hs.to_dual.smul` if you're changing `of_dual` to `to_dual` everywhere).

Reply:
The old proof doesn't work and the new proof is still a one-liner. I prefer having the same proofs for both the dual and the original lemma, unless the dualisation proof is shorter (which is not the case here anymore).